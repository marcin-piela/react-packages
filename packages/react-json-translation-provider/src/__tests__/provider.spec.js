import React from 'react';
import renderer from 'react-test-renderer';
import { FormattedMessage } from 'react-intl';

import createTranslationsProvider from '../provider';

describe('Translations provider factory', () => {
  test('throws an error if no locale is specified', () => {
    expect(() => {
      createTranslationsProvider();
    }).toThrowErrorMatchingSnapshot();
  });

  test('creates translations provider correctly', () => {
    const locales = {
      en: {
        FOO: {
          BAR: 'en.foobar',
        },
      },
    };

    const TranslationsProvider = createTranslationsProvider(locales);

    expect(
      renderer
        .create(
          <TranslationsProvider>
            <FormattedMessage id="FOO.BAR" />
          </TranslationsProvider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
});
