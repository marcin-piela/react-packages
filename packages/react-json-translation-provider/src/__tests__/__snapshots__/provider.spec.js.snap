// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Translations provider factory creates translations provider correctly 1`] = `
<span>
  en.foobar
</span>
`;

exports[`Translations provider factory throws an error if no locale is specified 1`] = `"You must provide at least one locale"`;
